language/move-vm/runtime/src/runtime.rs

```diff
@@ -49,7 +49,13 @@ impl VMRuntime {
         })
     }
 
+    pub(crate) fn invalidate_loader_cache(&self) {
+        self.loader.invalidate()
+    }
+
     pub fn new_session<'r, S: MoveResolver>(&self, remote: &'r S) -> Session<'r, '_, S> {
+        // If the loader cache is invalidated, flush it for the new session
+        self.loader.flush_if_invalidated();
```

Comment:
should we just call new_session_with_extensions here with NativeContextExtensions::default()?